File: test/node/disbursement/transfer_to_bank_inquiry_status_test.ts
import Dana from 'dana-node';
import { v4 as uuidv4 } from 'uuid';
import * as path from 'path';
import * as dotenv from 'dotenv';
import { getRequest, retryOnInconsistentRequest } from '../helper/util';
import { assertResponse } from '../helper/assertion';

dotenv.config();

const transferToBankTitleCase = 'TransferToBank';
const titleCase = 'TransferToBankInquiryStatus';
const jsonPathFile = path.resolve(__dirname, '../../../resource/request/components/Disbursement.json');

const dana = new Dana({
  partnerId: process.env.X_PARTNER_ID || '',
  privateKey: process.env.PRIVATE_KEY || '',
  origin: process.env.ORIGIN || '',
  env: process.env.ENV || 'sandbox',
});

async function createTransferToBankForInquiry(): Promise<string> {
  let partnerReferenceNo = '';
  await retryOnInconsistentRequest(async () => {
    partnerReferenceNo = uuidv4();
    const requestData: any = getRequest(jsonPathFile, transferToBankTitleCase, 'DisbursementBankValidAccount');
    requestData.partnerReferenceNo = partnerReferenceNo;
    await dana.disbursementApi.transferToBank(requestData);
  }, 3, 2000);
  return partnerReferenceNo;
}

describe('Disbursement - Transfer To Bank Inquiry Status Tests', () => {
  jest.retryTimes(2, { logErrorsBeforeRetry: true });

  test('TransferToBankInquiryStatusSuccessful - should successfully inquire transfer to bank status', async () => {
    const originalPartnerReferenceNo = await createTransferToBankForInquiry();
    const caseName = 'TransferToBankInquiryStatusSuccessful';
    const requestData: any = getRequest(jsonPathFile, titleCase, caseName);
    requestData.originalPartnerReferenceNo = originalPartnerReferenceNo;

    try {
      const response = await dana.disbursementApi.transferToBankInquiryStatus(requestData);
      await assertResponse(jsonPathFile, titleCase, caseName, response, {
        originalPartnerReferenceNo,
      });
    } catch (e) {
      console.error(`[REF] case=${caseName} originalPartnerReferenceNo=${originalPartnerReferenceNo}`);
      throw e;
    }
  });
});

File: test/java/src/test/java/id/dana/disbursement/TransferToBankInquiryStatusTest.java
package id.dana.disbursement;

import id.dana.disbursement.v1.api.DisbursementApi;
import id.dana.disbursement.v1.model.TransferToBankInquiryStatusRequest;
import id.dana.disbursement.v1.model.TransferToBankInquiryStatusResponse;
import id.dana.disbursement.v1.model.TransferToBankRequest;
import id.dana.disbursement.v1.model.TransferToBankResponse;
import id.dana.invoker.Dana;
import id.dana.invoker.model.DanaConfig;
import id.dana.invoker.model.constant.EnvKey;
import id.dana.invoker.model.enumeration.DanaEnvironment;
import id.dana.util.ConfigUtil;
import id.dana.util.RetryTestUtil;
import id.dana.util.TestUtil;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class TransferToBankInquiryStatusTest extends AbstractDisbursementTest {

  private static final Logger log = LoggerFactory.getLogger(TransferToBankInquiryStatusTest.class);

  private static final String transferToBankTitleCase = "TransferToBank";
  private static final String titleCase = "TransferToBankInquiryStatus";
  private static final String jsonPathFile = TransferToBankInquiryStatusTest.class.getResource(
      "/request/components/Disbursement.json").getPath();

  private DisbursementApi api;

  @BeforeEach
  void setUp() {
    DanaConfig.Builder danaConfigBuilder = new DanaConfig.Builder();
    danaConfigBuilder
        .partnerId(ConfigUtil.getConfig("X_PARTNER_ID", ""))
        .privateKey(ConfigUtil.getConfig("PRIVATE_KEY", ""))
        .origin(ConfigUtil.getConfig("ORIGIN", ""))
        .env(DanaEnvironment.getByName(ConfigUtil.getConfig(EnvKey.ENV, "SANDBOX")));

    DanaConfig.getInstance(danaConfigBuilder);
    api = Dana.getInstance().getDisbursementApi();
  }

  private String createTransferToBankForInquiry() throws IOException {
    String caseName = "DisbursementBankValidAccount";
    TransferToBankRequest requestData = TestUtil.getRequest(
        jsonPathFile, transferToBankTitleCase, caseName, TransferToBankRequest.class);

    String partnerReferenceNo = UUID.randomUUID().toString();
    log.info("[REF] case={} partnerReferenceNo={}", caseName, partnerReferenceNo);
    requestData.setPartnerReferenceNo(partnerReferenceNo);

    TransferToBankResponse response = api.transferToBank(requestData);
    if (response.getPartnerReferenceNo() != null && !response.getPartnerReferenceNo().isEmpty()) {
      return response.getPartnerReferenceNo();
    }
    return partnerReferenceNo;
  }

  @Test
  @RetryTestUtil.Retry(value = 3, waitMs = 2000)
  void testTransferToBankInquiryStatusSuccessful() throws IOException {
    String originalPartnerReferenceNo = createTransferToBankForInquiry();
    String caseName = "TransferToBankInquiryStatusSuccessful";

    TransferToBankInquiryStatusRequest requestData = TestUtil.getRequest(
        jsonPathFile, titleCase, caseName, TransferToBankInquiryStatusRequest.class);
    requestData.setOriginalPartnerReferenceNo(originalPartnerReferenceNo);

    Map<String, Object> variableDict = new HashMap<>();
    variableDict.put("originalPartnerReferenceNo", originalPartnerReferenceNo);

    TransferToBankInquiryStatusResponse response = api.transferToBankInquiryStatus(requestData);
    TestUtil.assertResponse(jsonPathFile, titleCase, caseName, response, variableDict);
  }
}
